src/lib/documentAnalysis/plugin-system/mixins/LocationTracking.ts

```diff
@@ -0,0 +1,178 @@
+/**
+ * Mixin for location tracking functionality
+ * Provides automatic location finding and line number calculation
+ */
+
+import { TextChunk } from '../TextChunk';
+import { Finding } from '../types';
+import { LocationUtils } from '../../utils/LocationUtils';
```

Comment:
Why is this a Mixin? Or, it's in the /mixins dir. I assuume it can be regular functions? Then maybe refactored just into utiltiies?